### PR TITLE
[INTEGRATION][COMMON] README.md to contain compatibility matrix table of integrations

### PR DESCRIPTION
**Pull request by howardyoo**
Created README md file in order to specify the latest version of the integrations as well as currently compatible version of which the OpenLineage is compatible with.

### Problem

As part of the work related to https://github.com/OpenLineage/OpenLineage/issues/602, I would like to submit this PR to create a mark down page under /integrations to summarize the current status of integrations in more concise way so that users would have access to the compatibility matrix of the supported integrations

Related to: #602 

### Solution

This matrix will provide information to users on which version OpenLineage integrations are last tested and compatible with.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)

Signed-off-by: Howard Yoo <howard.yoo@astronomer.io>

### CODE REVIEW

File: integration/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# OpenLineage Integrations
+|Integration|Release Link|Compatible Version|Latest Release|
+|-|-|-|-|
+|Apache Airflow|https://github.com/apache/airflow/releases|1.10.12+, 2.1+ (experimental)|2.2.5|
+|dagster|https://github.com/dagster-io/dagster/releases|0.13.8+|0.14.7|
+|dbt|https://github.com/dbt-labs/dbt-core/releases|0.20+|1.0.4|
+|spark|https://github.com/apache/spark/tags|3.2.1+|3.2.1|
```
Comment (pawel-big-lebowski) [inline integration/README.md:7]: Please refer to circle-ci config:
https://github.com/OpenLineage/OpenLineage/blob/main/.circleci/workflows/openlineage-java.yml#L12
for Spark versions tested in CI.

**Comment by howardyoo** [comment]
Hi all, I mistakenly put the signoff on the PR message, not in the commit message, so it's failing DCO test - I tried to fix it, but it seems like I don't have the access privilege to do it. Can someone fix this problem for me? Apologies (it was my first time doing the PR)

**Comment by collado-mike** [comment]
@howardyoo , you should be able to run the following from your local branch

```bash
git rebase --signoff origin/main
git push --force origin/patch-1
```
I'm assuming `origin` is what you have mapped to the `howardyoo` fork - possibly rename accordingly

**Comment by howardyoo** [comment]
Closing and re-doing this - as DCO check failed and looks like easiest way is to re-doing it.

**Pull request closed without merging**